--- a/build/gen.py
+++ b/build/gen.py
@@ -490,12 +490,6 @@
       'check_all',
       inputs=[
           ninja.RunBinary(
-              'check_formatter',
-              inputs=[ninja.source_file('tools/run_formatter.sh')],
-              implicit_inputs=ninja.directory(ninja.source_file('src'), ['target']),
-              args='--diff',
-          ),
-          ninja.RunBinary(
               'check_reference',
               inputs=[ninja.source_file('tools/update_reference.sh')],
               implicit_inputs=[
