--- a/k3screenctrl/files/k3screenctrl.init
+++ b/k3screenctrl/files/k3screenctrl.init
@@ -10,5 +10,5 @@ start() {
 }
 
 stop() {
-    killall k3screenctrl
+    killall -q `which k3screenctrl`
 }

--- a/luci-app-cpulimit/root/etc/init.d/cpulimit
+++ b/luci-app-cpulimit/root/etc/init.d/cpulimit
@@ -12,5 +12,5 @@ start() {
 }
 
 stop() {
-  killall -9 cpulimit
+  killall -q -9 `which cpulimit`
 }

--- a/luci-app-homebridge/files/root/etc/init.d/homebridge
+++ b/luci-app-homebridge/files/root/etc/init.d/homebridge
@@ -175,7 +175,7 @@ start_sub_service() {
 
 stop() {
 	echo "stop function" >> $LOG_FILE
-	killall -9 homebridge >> $LOG_FILE 2>&1
+	killall -9 `which homebridge` >> $LOG_FILE 2>&1
 #	config_load homebridge
 #	stop_main_service
 #	config_foreach stop_sub_service platform

--- a/luci-app-mentohust/root/etc/init.d/mentohust
+++ b/luci-app-mentohust/root/etc/init.d/mentohust
@@ -73,7 +73,7 @@ start()
 
 stop()
 {
-    killall mentohust
+    killall -q `which mentohust`
 }
 
 restart()

--- a/luci-app-pppoe-relay/root/etc/init.d/pppoe-relay
+++ b/luci-app-pppoe-relay/root/etc/init.d/pppoe-relay
@@ -25,5 +25,5 @@ start() {
 }
 
 stop() {
-	killall -9 pppoe-relay
+	killall -q -9 `which pppoe-relay`
 }

--- a/luci-app-pppoe-server/root/etc/init.d/pppoe-server
+++ b/luci-app-pppoe-server/root/etc/init.d/pppoe-server
@@ -97,5 +97,5 @@ stop() {
 		done
 	fi
 	ps | grep "pppd plugin rp-pppoe.so" | grep -v "grep" | awk '{print $1}' | xargs kill -9 >/dev/null 2>&1 &
-	killall -q -9 pppoe-server
+	killall -q -9 `which pppoe-server`
 }

--- a/sysuh3c/files/sysuh3c.init
+++ b/sysuh3c/files/sysuh3c.init
@@ -39,5 +39,5 @@ start()
 
 stop()
 {
-	killall sysuh3c >/dev/null 2>&1
+	killall -q `which sysuh3c`
 }

--- a/luci-app-speedtest-web/root/etc/init.d/speedtest-web
+++ b/luci-app-speedtest-web/root/etc/init.d/speedtest-web
@@ -15,5 +15,5 @@ start() {
 }
 
 stop() {
-	killall speedtest-web >/dev/null 2>&1
+	killall -q -9 `which speedtest-web`
 }
