--- a/luci-app-store/luasrc/controller/store.lua
+++ b/luci-app-store/luasrc/controller/store.lua
@@ -2,7 +2,7 @@ module("luci.controller.store", package.seeall)
 
 local myopkg = "is-opkg"
 local is_backup = "/usr/libexec/istore/backup"
-local page_index = {"admin", "store", "pages"}
+local page_index = {"admin", "system", "store"}
 
 function index()
     local function store_api(action, onlypost)
@@ -13,8 +13,7 @@ function index()
 
     local action
 
-    entry({"admin", "store"}, call("redirect_index"), _("iStore"), 31)
-    entry({"admin", "store", "pages"}, call("store_index")).leaf = true
+    entry({"admin", "system", "store"}, call("store_index"), _("iStore"), 31)
     if nixio.fs.access("/usr/lib/lua/luci/view/store/main_dev.htm") then
         entry({"admin", "store", "dev"}, call("store_dev")).leaf = true
     end
@@ -249,7 +248,7 @@ function store_action(param)
         local metapkg = pkg and (metapkgpre .. pkg) or ""
         if action == "update" or pkg then
             if action == "update" or action == "install" then
-                code, out, err = _action(myopkg, action, metapkg)
+                code, out, err = _action(myopkg, action .. " --force-overwrite --force-checksum --force-depends", metapkg)
             else
                 local meta = json_parse(fs.readfile(metadir .. "/" .. pkg .. ".json"))
                 local pkgs = {}
@@ -316,7 +315,7 @@ function store_upload()
         if string.lower(string.sub(path, -4, -1)) == ".run" then
             code, out, err = _action("sh", "-c", "ls -l \"%s\"; md5sum \"%s\" 2>/dev/null; chmod 755 \"%s\" && \"%s\"; RET=$?; rm -f \"%s\"; exit $RET" %{ path, path, path, path, path })
         else
-            code, out, err = _action("sh", "-c", "opkg install \"%s\"; RET=$?; rm -f \"%s\"; exit $RET" %{ path, path })
+            code, out, err = _action("sh", "-c", "opkg install --force-overwrite --force-checksum --force-depends \"%s\"; RET=$?; rm -f \"%s\"; exit $RET" %{ path, path })
         end
     else
         code = 500

--- a/luci-app-store/luasrc/view/store/main.htm
+++ b/luci-app-store/luasrc/view/store/main.htm
@@ -26,6 +26,8 @@ <h2 name="content"><%:iStore%>
     <a onclick="void(0)" href="https://github.com/linkease/istore/issues/22" target="_blank" style="text-decoration: none;">
         v<%=id.version%>
     </a>
+	<br/>
+	<span style="font-weight:normal;font-size:70%; color:orangered">更多插件, 请使用 <a href="/cgi-bin/luci/admin/system/opkg" style="text-decoration: none;">系统->软件包 </span>
 </h2>
 <link rel="stylesheet" href="/luci-static/istore/style.css?v=<%=id.version%>">
 <div id="app">
@@ -58,4 +60,4 @@ <h2 name="content"><%:iStore%>
 </script>
 
 <script type="module" crossorigin src="/luci-static/istore/index.js?v=<%=id.version%>"></script>
-<%+footer%>
\ No newline at end of file
+<%+footer%>
