--- a/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js
+++ b/luci-app-samba4/htdocs/luci-static/resources/view/samba4.js
@@ -113,6 +113,10 @@ return view.extend({
 		o.rmempty = false;
 
 		s.option(form.Flag, 'force_root', _('Force Root'));
+		o.enabled = 'yes';
+		o.disabled = 'no';
+		o.default = 'yes';
+		o.rmempty = false;
 
 		o = s.option(form.Value, 'users', _('Allowed users'));
 		o.rmempty = true;
