From 68c28fb85cf079a678ce5334a18b9d848faa9d06 Mon Sep 17 00:00:00 2001
From: mzz2017 <mzz@tuta.io>
Date: Sun, 19 Dec 2021 17:08:13 +0800
Subject: [PATCH] fix: a problem that REDIRECT must work with port-sharing for
 intranet

---
 service/core/v2ray/v2rayTmpl.go | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/core/v2ray/v2rayTmpl.go
+++ b/core/v2ray/v2rayTmpl.go
@@ -878,6 +878,13 @@ func (t *Template) setDualStack() {
 	if !t.Setting.PortSharing {
 		// copy a group of ipv6 inbounds and set the tag
 		for i := range t.Inbounds {
+			if t.Inbounds[i].Tag == "transparent" && t.Setting.TransparentType == configure.TransparentRedirect {
+				// https://ipset.netfilter.org/iptables-extensions.man.html#lbDK
+				// REDIRECT redirects the packet to the machine itself by changing the destination IP to the primary address of the incoming interface.
+				// So we should listen at 0.0.0.0 instead of 127.0.0.1
+				inbounds6[i].Tag = "THIS_IS_A_DROPPED_TAG"
+				continue
+			}
 			if t.Inbounds[i].Tag == "dns-in" {
 				t.Inbounds[i].Listen = "127.2.0.17"
 				inbounds6[i].Tag = "THIS_IS_A_DROPPED_TAG"
