--- a/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js
+++ b/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js
@@ -69,6 +69,11 @@ return view.extend({
 			p[i].value('ACCEPT', _('accept'));
 		}
 
+		if (L.hasSystemFeature('fullconenat')) {
+		o = s.option(form.Flag, 'fullcone', _('FullCone NAT'), _('Using FullCone NAT can improve gaming performance effectively'));
+		o = s.option(form.Flag, 'fullcone6', _('Enable FullCone NAT6'));
+		}
+
 		/* Netfilter flow offload support */
 
 		if (L.hasSystemFeature('offloading')) {
