--- a/coremark/Makefile
+++ b/coremark/Makefile
@@ -48,14 +48,14 @@ define Package/coremark/config
 	config COREMARK_ENABLE_MULTITHREADING
 		bool "Enable multithreading support"
 		depends on PACKAGE_coremark
-		default n
+		default y
 		help
 			This enables multithreading support
 
 	config COREMARK_NUMBER_OF_THREADS
 		int "Number of threads"
 		depends on COREMARK_ENABLE_MULTITHREADING
-		default 2
+		default 10
 		help
 			Number of threads to run in parallel
 endef
