--- a/luci-app-ddns/Makefile
+++ b/luci-app-ddns/Makefile
@@ -12,7 +12,7 @@ PKG_LICENSE:=Apache-2.0
 PKG_MAINTAINER:=Ansuel Smith <ansuelsmth@gmail.com>
 
 LUCI_TITLE:=LuCI Support for Dynamic DNS Client (ddns-scripts)
-LUCI_DEPENDS:=+luci-mod-admin-full +ddns-scripts
+LUCI_DEPENDS:=+luci-mod-admin-full +ddns-scripts +ddns-scripts-aliyun +ddns-scripts-dnspod +ddns-scripts-cloudflare
 LUCI_PKGARCH:=all
 
 include ../../luci.mk

--- a/ddns-scripts/Makefile
+++ b/ddns-scripts/Makefile
@@ -115,12 +115,22 @@ endef
 
 define Package/ddns-scripts-dnspod
   $(call Package/ddns-scripts/Default)
-  TITLE:=Extension for dnspod.cn API
-  DEPENDS:=ddns-scripts +curl
+  TITLE:=Extension for DNSPod API
+  DEPENDS:=ddns-scripts +curl +bind-host +openssl-util
 endef
 
 define Package/ddns-scripts-dnspod/description
-  Dynamic DNS Client scripts extension for dnspod.cn API (require curl)
+  Dynamic DNS Client scripts extension for Tencent Cloud API V3 (require curl)
+endef
+
+define Package/ddns-scripts-aliyun
+  $(call Package/ddns-scripts/Default)
+  TITLE:=Extension for aliyun.com API
+  DEPENDS:=ddns-scripts +curl +bind-host +openssl-util
+endef
+
+define Package/ddns-scripts-aliyun/description
+  Dynamic DNS Client scripts extension for aliyun.com API (require curl)
 endef
 
 
@@ -326,7 +336,8 @@ define Package/ddns-scripts-services/install
 	rm $(1)/usr/share/ddns/default/freedns.42.pl.json
 	rm $(1)/usr/share/ddns/default/godaddy.com-v1.json
 	rm $(1)/usr/share/ddns/default/digitalocean.com-v2.json
-	rm $(1)/usr/share/ddns/default/dnspod.cn.json
+	rm $(1)/usr/share/ddns/default/dnspod-tencent-cloud.json
+	rm $(1)/usr/share/ddns/default/aliyun.com.json
 	rm $(1)/usr/share/ddns/default/no-ip.com.json
 	rm $(1)/usr/share/ddns/default/bind-nsupdate.json
 	rm $(1)/usr/share/ddns/default/route53-v1.json
@@ -421,7 +432,7 @@ define Package/ddns-scripts-dnspod/install
 		$(1)/usr/lib/ddns
 
 	$(INSTALL_DIR) $(1)/usr/share/ddns/default
-	$(INSTALL_DATA) ./files/usr/share/ddns/default/dnspod.cn.json \
+	$(INSTALL_DATA) ./files/usr/share/ddns/default/dnspod-tencent-cloud.json \
 		$(1)/usr/share/ddns/default/
 endef
 
@@ -434,6 +445,25 @@ exit 0
 endef
 
 
+define Package/ddns-scripts-aliyun/install
+	$(INSTALL_DIR) $(1)/usr/lib/ddns
+	$(INSTALL_BIN) ./files/usr/lib/ddns/update_aliyun_com.sh \
+		$(1)/usr/lib/ddns
+
+	$(INSTALL_DIR) $(1)/usr/share/ddns/default
+	$(INSTALL_DATA) ./files/usr/share/ddns/default/aliyun.com.json \
+		$(1)/usr/share/ddns/default/
+endef
+
+define Package/ddns-scripts-aliyun/prerm
+#!/bin/sh
+if [ -z "$${IPKG_INSTROOT}" ]; then
+	/etc/init.d/ddns stop
+fi
+exit 0
+endef
+
+
 define Package/ddns-scripts-noip/install
 	$(INSTALL_DIR) $(1)/usr/lib/ddns
 	$(INSTALL_BIN) ./files/usr/lib/ddns/update_no-ip_com.sh \
@@ -612,6 +642,7 @@ $(eval $(call BuildPackage,ddns-scripts-freedns))
 $(eval $(call BuildPackage,ddns-scripts-godaddy))
 $(eval $(call BuildPackage,ddns-scripts-digitalocean))
 $(eval $(call BuildPackage,ddns-scripts-dnspod))
+$(eval $(call BuildPackage,ddns-scripts-aliyun))
 $(eval $(call BuildPackage,ddns-scripts-noip))
 $(eval $(call BuildPackage,ddns-scripts-nsupdate))
 $(eval $(call BuildPackage,ddns-scripts-route53))