--- a/luci-app-nginx-manager/Makefile
+++ b/luci-app-nginx-manager/Makefile
@@ -6,7 +6,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=LuCI support for NginxManager From sundaqiang
-LUCI_DEPENDS:=+luci-nginx +luci-ssl-nginx +luci-ssl-openssl
+LUCI_DEPENDS:=+luci-ssl-nginx
 LUCI_PKGARCH:=all
 PKG_VERSION:=1.0
 PKG_RELEASE:=20211030

--- a/luci-app-nginx-manager/luasrc/controller/nginx-manager.lua
+++ b/luci-app-nginx-manager/luasrc/controller/nginx-manager.lua
@@ -14,9 +14,9 @@ function index()
     end
     nixio.fs.writefile("/etc/config/nginx-manager", "")
     x = luci.model.uci.cursor()
-    x:set("nginx-manager", "main", "nginx")
-    x:set("nginx-manager", "main", "name", "main")
-    x:set("nginx-manager", "main", "filepath", "/etc/nginx/nginx.conf")
+    x:set("nginx-manager", "config", "nginx")
+    x:set("nginx-manager", "config", "name", "config")
+    x:set("nginx-manager", "config", "filepath", "/etc/config/nginx")
     for path in nixio.fs.dir("/etc/nginx/conf.d") do
         if path:find(".conf$") ~= nil then
             name = path:gsub(".conf", "")

--- a/luci-app-nginx-manager/luasrc/view/nginx-manager/index.htm
+++ b/luci-app-nginx-manager/luasrc/view/nginx-manager/index.htm
@@ -24,9 +24,6 @@
 				<input type="submit" name="cbi.rts.<%=self.config%>.<%=k%>" onclick="this.form.cbi_state='del-section'; return true" value="<%:Delete%>" class="cbi-button" />
 			</div>
 		<%- end %>
-
-		<%+cbi/tabmenu%>
-
 		<fieldset class="cbi-section-node<% if self.tabs then %> cbi-section-node-tabbed<% end %>" id="cbi-<%=self.config%>-<%=section%>" style="border-bottom: 1px solid #ddd;border-radius: 0px">
 			<%+cbi/ucisection%>
 		</fieldset>

--- a/luci-app-nginx-manager/luasrc/model/cbi/nginx-manager.lua
+++ b/luci-app-nginx-manager/luasrc/model/cbi/nginx-manager.lua
@@ -1,5 +1,5 @@
 local fs = require "nixio.fs"
-local m = Map("nginx-manager",translate("Nginx Manager"), translate("A simple Nginx manager") .. [[<br /><br /><a href="https://github.com/sundaqiang/openwrt-packages" target="_blank">Powered by sundaqiang</a>]])
+local m = Map("nginx-manager",translate("Nginx Manager"), translate("A simple Nginx manager") .. [[<br /><a href="https://github.com/sundaqiang/openwrt-packages" target="_blank">Powered by sundaqiang</a>]])
 s = m:section(TypedSection, "nginx", translate("Web site list"))
 s.template = "nginx-manager/index"
 s.addremove = true
