--- a/luci-app-advanced/Makefile
+++ b/luci-app-advanced/Makefile
@@ -35,9 +35,6 @@ define Package/$(PKG_NAME)/install
 	
 	$(INSTALL_DIR) $(1)/etc/uci-defaults
 	$(INSTALL_BIN) ./root/etc/uci-defaults/* $(1)/etc/uci-defaults/
-	
-	$(INSTALL_DIR) $(1)/bin
-	$(INSTALL_BIN) ./root/bin/* $(1)/bin/
 endef
 
 $(eval $(call BuildPackage,$(PKG_NAME)))

--- a/luci-app-advanced/luasrc/model/cbi/advanced.lua
+++ b/luci-app-advanced/luasrc/model/cbi/advanced.lua
@@ -140,6 +140,28 @@ end
 end
 end
 
+if nixio.fs.access("/etc/config/nginx")then
+s:tab("nginxconf",translate("NGINX"),translate("本页是配置/etc/config/nginx包含NGINX配置文档内容。应用保存后自动重启生效"))
+conf=s:taboption("nginxconf",Value,"nginxconf",nil,translate("开头的数字符号（＃）或分号的每一行（;）被视为注释；删除（;）启用指定选项。"))
+conf.template="cbi/tvalue"
+conf.rows=20
+conf.wrap="off"
+conf.cfgvalue=function(t,t)
+return e.readfile("/etc/config/nginx") or ""
+end
+conf.write=function(a,a,t)
+if t then
+t=t:gsub("\r\n?","\n")
+e.writefile("/tmp/nginx",t)
+if(luci.sys.call("cmp -s /tmp/nginx /etc/config/nginx")==1)then
+e.writefile("/etc/config/nginx",t)
+luci.sys.call("/etc/init.d/nginx reload >/dev/null")
+end
+e.remove("/tmp/nginx")
+end
+end
+end
+
 if nixio.fs.access("/etc/config/mwan3")then
 s:tab("mwan3conf",translate("负载均衡"),translate("本页是配置/etc/config/mwan3包含负载均衡设置文档内容。应用保存后自动重启生效"))
 conf=s:taboption("mwan3conf",Value,"mwan3conf",nil,translate("开头的数字符号（＃）或分号的每一行（;）被视为注释；删除（;）启用指定选项。"))
