From 45368856980e030ddd95221289304c33e2749aa6 Mon Sep 17 00:00:00 2001
From: Beginner-Go <70857188+Beginner-Go@users.noreply.github.com>
Date: Mon, 26 Feb 2024 15:45:35 +0000
Subject: [PATCH] fix go 1.21 build error

---
 go.mod | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/go.mod b/go.mod
index 71a9a02..077ac79 100644
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/teddysun/xray-plugin
 
-go 1.22
+go 1.21.5
 
 require (
 	github.com/xtls/xray-core v1.8.8
-- 
2.34.1

